string.c

```diff
@@ -2873,15 +2873,29 @@ rb_str_append(VALUE str, VALUE str2)
     return rb_str_buf_append(str, str2);
 }
 
+#define MIN_PRE_ALLOC_SIZE 48
+
 VALUE
 rb_str_concat_literals(size_t num, const VALUE *strary)
 {
     VALUE str;
-    size_t i;
+    size_t i, s;
+    long len = 1;
+
+    if (UNLIKELY(!num)) return rb_str_new(0, 0);
+    if (UNLIKELY(num == 1)) return rb_str_resurrect(strary[0]);
+
+    for (i = 0; i < num; ++i) { len += RSTRING_LEN(strary[i]); }
+    if (LIKELY(len < MIN_PRE_ALLOC_SIZE)) {
+      str = rb_str_resurrect(strary[0]);
+      s = 1;
+    }
+    else {
+      str = rb_str_buf_new(len);
```

Comment:
I fixed it by https://github.com/ruby/ruby/pull/1626/commits/1f6ad9bf87ba3dd6a9e0ce179ce41348056b6bd4 ! 💡 

I referenced the same operation in array.c. https://github.com/ruby/ruby/blob/trunk/array.c#L1974